Peer dependencies pinned to React 18.2 incompatible with latest nextjs

### Issue Summary

[@calcom/embed-react has peer dependencies pinned to React 18.2](https://github.com/calcom/cal.com/blob/main/packages/embeds/embed-react/package.json#L32-L34) but nextjs assumes React 19.

Although one can still install the package using `npm install --force,` some deployment services, e.g., AWS Amplify, fail to build the web app because of version incompatibility.

### Steps to Reproduce

1. Setup a new nextjs project using `npx create-next-app@latest`
2. Try to install `@calcom/embed-react` via `npm install @calcom/embed-react`

### Actual Results

```
2024-12-28T15:31:23.922Z [WARNING]: ERR! Found: react@19.0.0
npm ERR! node_modules/react
npm ERR! react@"^19.0.0" from the root project
npm ERR! peer react@">=16.8.0" from @floating-ui/react@0.26.28
npm ERR! node_modules/@floating-ui/react
npm ERR! @floating-ui/react@"^0.26.16" from @headlessui/react@2.2.0
npm ERR! node_modules/@headlessui/react
npm ERR! @headlessui/react@"^2.2.0" from the root project
npm ERR! 13 more (@floating-ui/react-dom, @headlessui/react, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^18.2.0" from @calcom/embed-react@1.5.2
npm ERR! node_modules/@calcom/embed-react
npm ERR!
2024-12-28T15:31:23.922Z [WARNING]: @calcom/embed-react@"^1.5.2" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@18.3.1
npm ERR! node_modules/react
npm ERR! peer react@"^18.2.0" from @calcom/embed-react@1.5.2
npm ERR! node_modules/@calcom/embed-react
npm ERR! @calcom/embed-react@"^1.5.2" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
2024-12-28T15:31:23.922Z [WARNING]: npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! /codebuild/output/src3840928023/src/qproductlab.com/.npm/_logs/2024-12-28T15_31_23_198Z-eresolve-report.txt
```

### Expected Results

No errors during installation and build stage.

--- a/packages/embeds/embed-react/package.json
+++ b/packages/embeds/embed-react/package.json
@@ -30,8 +30,8 @@
   "module": "./dist/Cal.es.mjs",
   "types": "./dist/embed-react/src/index.d.ts",
   "peerDependencies": {
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react": "^18.2.0 || ^19.0.0",
+    "react-dom": "^18.2.0 || ^19.0.0"
   },
   "files": [
     "dist"

